### DISCUSSION TITLE
Component Inside table slot:body not being sorted

### DISCUSSION THREAD

**Discussion by RamonDonadeu** [quasarframework/quasar#15484] [General - Components / Directives / etc] [answered]
I can't do a CodePen because I don't know how to add components on it but i'll share photos and i Know were the error is but i don't know how to solve it.

I have a complex Table structure with custom templates. Some of columns of the table are editable, but, because there are dates, booleans... in my table, I need to format and personalize the display of this kind of data.

I get the date on format dd/mm/yyyy and qinput type="date" needs yyyy-mm-dd so I need to format that data to desplay it, update it, and send it to the backend.

So, i made a component called CommonTableValue:

```html
<template>
  <span>
    {{ editValue }}
  </span>
  <!-- Some code here -->
  <!-- POPUP EDIT -->
  <q-popup-edit
    v-model="formatedValue"
    :title="col.label"
    v-if="col.editable"
    v-slot="scope"
    buttons
    @save=" (val, initialValue) => updateTableValue(val, initialValue, props.row, col)"
  >

    <q-input
      v-model="scope.value"
      v-if="col.type === 'date'"
      type="date"
      dense
      autofocus
    />
    <q-input
      v-model="scope.value"
      v-if="col.type === 'number'"
      type="number"
      dense
      autofocus
    />
    <q-input
      v-model="scope.value"
      v-if="col.type === 'string'"
      type="text"
      dense
      autofocus
    />
  </q-popup-edit>
  <!-- Some code here -->
</template>
<script>
export default {
  name: 'CommonTableValue',
  data () {
    return {
      editValue: null,
      formatedValue: null
    }
  },
  props: {
    col: { type: Object, required: true },
    props: { type: Object, required: true },
    pageActions: { type: Object, required: true }
  },
  emits: ['updateTableValue'],
  methods: {
    transformDate (val) {
      this.editValue = val.split('-').reverse().join('/')
    },
    updateTableValue (value, initValue, row, col) {
      if (col.type === 'date') {
        this.transformDate(value)
      } else {
        this.editValue = value
      }
      this.$emit('updateTableValue', {
        newValue: value,
        updateFunction: col.editFunction,
        row,
        col
      })
    }
  },
  mounted () {
    this.editValue = this.col.value
    if (this.col.type === 'date') {
      this.formatedValue = this.props.row[this.col.name]
        .split('/')
        .reverse()
        .join('-')
    } else if (this.col.type === 'number') {
      this.formatedValue = parseInt(this.props.row[this.col.name])
    }
  }
}
</script>

```

This works fine aparently, The diferent types are displayed and formated correctly and sends the update to the backend through emits to the main component/layout

```html
 <template v-slot:body="props">
      <q-tr :props="props">
        <!-- Some code here -->
        <q-td
          v-for="col in props.cols"
          :key="col.name"
          :props="props"
          @click="onRowClick(props.row)"
        >
          <!-- IF there aren't any special options, show value -->
          <CommonTableValue
            :props="props"
            :col="col"
            :page-actions="pageActions"
            @updateTableValue="updateTableValue"
          ></CommonTableValue>
          
        <!-- Some code here -->
        </q-td>
      </q-tr>
    </template>

```
But, I found out that, when i pressed the column title to sort the column by it's content. The rows weren't being sorted. Visually, because they get sorted but not visible.
![Captura de pantalla 2023-02-23 a las 16 32 12](https://user-images.githubusercontent.com/43962562/220954482-aba5579a-8fea-45d1-a96e-1f245fbf8a16.png) ![Captura de pantalla 2023-02-23 a las 16 32 29](https://user-images.githubusercontent.com/43962562/220954500-04461dbf-47d1-4ae6-bb45-07c8df08bfb0.png) ![Captura de pantalla 2023-02-23 a las 16 32 33](https://user-images.githubusercontent.com/43962562/220954540-4948bd03-7182-4185-bcd4-61495c4273bc.png)

I found out that my component "CommonTableValue" wasn't being rerendered when rows where sorted. And, if I add the value of the column before "CommonTableValue" that value IS BEING SORTED, but the data inside my component stays the same.

```html
 <template v-slot:body="props">
      <q-tr :props="props">
        <!-- Some code here -->
        <q-td
          v-for="col in props.cols"
          :key="col.name"
          :props="props"
          @click="onRowClick(props.row)"
        >
          <span>
            {{ col.value }}
          </span>
          <!-- IF there aren't any special options, show value -->
          <CommonTableValue
            :props="props"
            :col="col"
            :page-actions="pageActions"
            @updateTableValue="updateTableValue"
          ></CommonTableValue>
          
        <!-- Some code here -->
        </q-td>
      </q-tr>
    </template>
```

![Captura de pantalla 2023-02-23 a las 16 30 43](https://user-images.githubusercontent.com/43962562/220954066-c39fbf09-343d-4c29-981d-98072d5e6582.png) ![Captura de pantalla 2023-02-23 a las 16 33 37](https://user-images.githubusercontent.com/43962562/220954788-a9acf06e-41f8-46be-9698-598c06116190.png) ![Captura de pantalla 2023-02-23 a las 16 33 40](https://user-images.githubusercontent.com/43962562/220954812-25a7138b-cfee-4b8c-92bf-ed8de276a8d3.png)

Does Anybody Know where the problem is? or how to make it work? Thanks :)

**Comment by metalsadman** [comment, marked as answer]
table sorts what the value is for column `field` or `format` props.
